### PR TITLE
merged all CM into 1

### PR DESCRIPTION
**Pull request by Akarshit**
* A new class ContributionTab contains the stuff that is visible in a tab.
* ContributionManagerDialog contains the tabbedPane which has ContributionTabs
* ContribtionListing object is present in every ContributionTab object but it will be removed later to include a single object in ContributionManagerDialog so we can get rid of synchronization issues.
*  Added shortcuts to the tabs like Alt+1, Alt+2 on linux and windows etc.

waiting for @alignedleft to give this PR a go then @benfry can review and merge it.

merged this with the latest and tried to minimize the changes.

### CODE REVIEW

File: app/src/processing/app/contrib/ContributionManagerDialog.java [hunk 1]
```diff
@@ -410,36 +404,35 @@ public void actionPerformed(ActionEvent actionEvent) {
       public void keyPressed(KeyEvent e) {
         //System.out.println(e);
         KeyStroke wc = Toolkit.WINDOW_CLOSE_KEYSTROKE;
-        if ((e.getKeyCode() == KeyEvent.VK_ESCAPE) ||
-            (KeyStroke.getKeyStrokeForEvent(e).equals(wc))) {
+        if ((e.getKeyCode() == KeyEvent.VK_ESCAPE)
```
Comment (benfry) [inline app/src/processing/app/contrib/ContributionManagerDialog.java]: That was the correct formatting before you changed it: 
https://github.com/processing/processing/wiki/Style-Guidelines
Please read the guidelines.

File: app/src/processing/app/contrib/ContributionManagerDialog.java [hunk 2]
```diff
@@ -410,36 +404,35 @@ public void actionPerformed(ActionEvent actionEvent) {
...
-
-
-  protected void updateContributionListing() {
+*/
+  /*protected void updateContributionListing() {
     if (editor != null) {
-      List<Contribution> contributions = new ArrayList<Contribution>();
+      ArrayList<Contribution> contributions = new ArrayList<Contribution>();
 
-      List<Library> libraries =
```
Comment (benfry) [inline app/src/processing/app/contrib/ContributionManagerDialog.java]: This formatting was correct; do not change it.

File: app/src/processing/app/contrib/ContributionTab.java [hunk 3]
```diff
@@ -0,0 +1,589 @@
...
+
+    // Adding both of these to EAST shouldn't pose too much of a problem,
+    // since they can never get added together.
+      statusRestartPane.add(restartButton, BorderLayout.EAST);
+      statusRestartPane.add(retryConnectingButton, BorderLayout.EAST);
+
+      pane.add(statusRestartPane, BorderLayout.SOUTH);
+
+
+//      status = new StatusPanel();
```
Comment (benfry) [inline app/src/processing/app/contrib/ContributionTab.java]: For the future, just use /* and */ to remove large sections like this.

File: app/src/processing/app/contrib/ContributionManagerDialog.java (hunk 1 above)
Comment (Akarshit) [inline app/src/processing/app/contrib/ContributionManagerDialog.java]: @benfry i am using formatter.xml from processing isin't it supposed to take care of all this stuff ?

Comment (benfry) [inline app/src/processing/app/contrib/ContributionManagerDialog.java]: Some of the changes will still be subjective so a formatter isn't going to fix everything.

**Pull request merged**